File: core/src/com/alta_v2/game/AltaV2.java
package com.alta_v2.game;

import com.alta_v2.game.inputProcessor.InputProcessorFactory;
import com.alta_v2.mediatorModule.ProcessMediator;
import com.alta_v2.model.MenuDefinitionModel;
import com.alta_v2.mediatorModule.serde.ActionListener;
import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.InputProcessor;
import com.google.inject.Inject;

public class AltaV2 extends Game {

	private final ProcessMediator processMediator;
	private final InputProcessorFactory inputProcessorFactory;
	private InputProcessor inputProcessor;

	/**
	 * Initialize new instance of {@link AltaV2}.
	 */
	@Inject
    public AltaV2(InputProcessorFactory inputProcessorFactory,
				  ProcessMediator processMediator,
				  ScreenManager screenManager) {
		this.inputProcessorFactory = inputProcessorFactory;
		this.processMediator = processMediator;
		screenManager.setGame(this);
	}

	/**
	 * {@inheritDoc}
	 */
    @Override
	public void create () {
		Gdx.input.setInputProcessor(this.inputProcessor);
		this.processMediator.loadMenuScreen(new MenuDefinitionModel());
	}

	/**
	 * {@inheritDoc}
	 */
	@Override
	public void dispose() {
		System.exit(0);
	}

	public void setInputListener(ActionListener actionListener) {
		this.inputProcessor = this.inputProcessorFactory.createDefaultInputProcessor(actionListener);
	}
}

File: core/src/com/alta_v2/game/AltaV2.kt
package com.alta_v2.game

import com.alta_v2.game.inputProcessor.InputProcessorFactory
import com.alta_v2.mediator.ProcessMediator
import com.alta_v2.mediator.serde.ActionListener
import com.alta_v2.model.MenuDefinitionModel
import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.InputProcessor
import com.google.inject.Inject
import kotlin.system.exitProcess

class AltaV2 @Inject constructor(private val inputProcessorFactory: InputProcessorFactory,
                                 private val processMediator: ProcessMediator,
                                 screenManager: ScreenManager) : Game() {

    private var inputProcessor: InputProcessor? = null

    init {
        screenManager.setGame(this)
    }

    /**
     * {@inheritDoc}
     */
    override fun create() {
        Gdx.input.inputProcessor = inputProcessor
        processMediator.loadMenuScreen(MenuDefinitionModel())
    }

    /**
     * {@inheritDoc}
     */
    override fun dispose() {
        exitProcess(0)
    }

    fun setInputListener(actionListener: ActionListener) {
        inputProcessor = inputProcessorFactory.createDefaultInputProcessor(actionListener)
    }
}